### ISSUE TITLE
Omit sending optional arguments in query if variable is not provided

### ISSUE THREAD

**Issue by seanes** [apollographql/apollo-client#1276]
Disclaimer: At the very best it's a feature suggestion or a simple cry for help trying to solve something that at a surface scratch sounds overly complicated to do with GraphQL.

What I would like to do is create a query given a set of parameters (e.g. one required and _n_ optional), where the undefined optional arguments are _not_ passed as part of query to the server. 

Simple example. A search for stop places with optional filters provided:

```javascript
const findStop = gql`
    query findStop($query: String!, $stopPlaceType: stopPlaceType) {
        stopPlace(query: $query, stopPlaceType: [$stopPlaceType], size: 7) {
            id
        }
    },
`
```

Simply put, if `$stopPlaceType` is not provided as a variable, the argument should be left out in the `query findStop(...)`, as it's left out as as variable in the payload. 

Or is this simply something that should be left alone to the server to deal with? Following [this](https://github.com/graphql/graphql-js/issues/210) discussion it seems to be discouraged.

**Comment by helfer** [comment]
@seanes I'm not sure I understand the problem. If the arguments are optional, you should be able to simply not provide them without altering the query. The server will deal with it.

**Comment by seanes** [comment]
The problem was that the server wasn't allowing this in a fashionable matter. Closing this since it's unrelated. Thank you!

**Comment by chris08002** [comment]
@seanes Have you found a solution for this? I am not able to make a query with optional variables in angular apollo.
